javalib/src/main/scala/java/math/Conversion.scala

```diff
@@ -0,0 +1,513 @@
+      var v = highDigit & 0xFFFFFFFFL
+      if (sign < 0)
+        v = -v
+      java.lang.Long.toString(v, radix)
+    } else if (radix == 10 || radixOutOfBounds) {
+      bi.toString
+    } else {
+      var bitsForRadixDigit: Double = 0.0
+      bitsForRadixDigit = Math.log(radix) / Math.log(2)
+      val addForSign = if (sign < 0) 1 else 0
+      val biAbsLen = bi.abs().bitLength()
+      val resLenInChars = (biAbsLen / bitsForRadixDigit + addForSign).toInt + 1
+      val result = new Array[Char](resLenInChars)
+      var currentChar = resLenInChars
+      var resDigit: Int = 0
+
+      if (radix != 16) {
+        val temp = new Array[Int](numberLength)
+        System.arraycopy(digits, 0, temp, 0, numberLength)
+        var tempLen = numberLength
+        val charsPerInt = digitFitInInt(radix)
+        //var i: Int = 0
+        val bigRadix = bigRadices(radix - 2)
+
+        @inline
+        @tailrec
+        def loop(): Unit = {
+          resDigit = Division.divideArrayByInt(temp, temp, tempLen, bigRadix)
+          val previous = currentChar
+
+          @inline
+          @tailrec
+          def innerLoop(): Unit = {
+            currentChar -= 1
+            result(currentChar) = Character.forDigit(resDigit % radix, radix)
+            resDigit /= radix
+            if(resDigit != 0 && currentChar != 0)
+              innerLoop()
+          }
+          innerLoop()
+
+          val delta = charsPerInt - previous + currentChar
+          var i: Int = 0
+          while (i < delta && currentChar > 0) {
+            currentChar -= 1
+            result(currentChar) = '0'
+            i += 1
+          }
+          i = tempLen - 1
+          while (i > 0 && temp(i) == 0) {
+            i -= 1
+          }
+          tempLen = i + 1
+          if (!(tempLen == 1 && temp(0) == 0))
+            loop()
+        }
+
+        loop()
+
+      } else {
+        for (i <- 0 until numberLength) {
+          var j = 0
+          while (j < 8 && currentChar > 0) {
+            resDigit = digits(i) >> (j << 2) & 0xf
+            currentChar -= 1
+            result(currentChar) = java.lang.Character.forDigit(resDigit, 16)
+            j += 1
+          }
+        }
+      }
+      while (result(currentChar) == '0') {
+        currentChar += 1
+      }
+      if (sign == -1) {
+        currentChar -= 1
+        result(currentChar) = '-'
+      }
+      new String(result, currentChar, resLenInChars - currentChar)
+    }
+  }
+
+
+  /**
+   * Builds the correspondent {@code String} representation of {@code val} being
+   * scaled by {@code scale}.
+   *
+   * @see BigInteger#toString()
+   * @see BigDecimal#toString()
+   */
+  def toDecimalScaledString(bi: BigInteger, scale: Int): String = {
+
+    val sign:Int = bi.sign
+    val numberLength:Int = bi.numberLength
+    val digits:Array[Int] = bi.digits
+    var resLengthInChars: Int = 0
+    var currentChar: Int = 0
+
+    if (sign == 0) scale match {
+      case 0 => return "0"
```

Comment:
You can avoid all those `return`s by using an `else` with this `if (sign == 0)`.